.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,55 @@
+name: CodeQL Analysis
+
+on:
+  push:
+    # ignore dependabot branches on push -> https://github.com/microsoft/binskim/issues/425#issuecomment-893373709
+    branches-ignore:
+      - 'dependabot/**'
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: windows-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          queries: security-and-quality
+          languages: csharp
+
+      - name: Build solution
+        shell: pwsh
+        run: |
+          $pathToSolution = "ApiVersioning.sln"
+          $buildConfiguration = "Release"
+          $useSharedCompilation = "false"
+          $testProjects = @( Get-ChildItem -Path test/ -Recurse | Where-Object { $_.Extension -match 'shproj' -or $_.Extension -match 'csproj' } | ForEach-Object { $_.FullName } )
+
+          dotnet nuget locals all --clear
+
+          # remove one or more test projects, 
+          # so that CodeQL only analyzes the source code
+          dotnet sln $pathToSolution remove $testProjects
```

Comment:
This is a bad idea. This makes modifications to the Solution (`*.sln`) file. Yes, it can be undone with source control or thrown away in a temporary build, but that is not intrinsically obvious to someone testing this locally. There are at least two alternatives.

### Alterative 1 : Loop Over Target Projects
I just realized that the query above is filtering out test projects. Inverting the query would allow looping over the target projects without any modification.  For example:

```powershell
Get-ChildItem -Path src -Include *.csproj -Recurse | ForEach-Object { <# use $_.FullName #> }
```

### Alternative 2 : Use Solution Filters
The simpler and _saner_ approach would be to use [Solution Filters](https://docs.microsoft.com/en-us/visualstudio/msbuild/solution-filters) (`*.slnf`). I presume this would be called something like `codeql-analysis.slnf` and only include the projects to be analyzed. All `dotnet` commands would then accept the `*.slnf` file instead of the `*.sln` file.

> NOTE: `*.slnf` files have been natively supported by MSBuild since 16.7 (see [Issue 4097](https://github.com/dotnet/msbuild/issues/4097))